### PR TITLE
[#1011] Dark mode

### LINKED ISSUE #1011: Enable dark mode
## Is your feature request related to a problem? Please describe.
Once we decide to support dark app mode, we need to check and possibly enhance the UI of all the apps's screens and enable it with the Gradle `IS_DARK_MODE_ENABLED` option.  

## Describe the solution you'd like
- Electric-Coin-Company/zashi#32
- `gradle.properties#IS_DARK_MODE_ENABLED = true`

### LINKED ISSUE #1337: Redesign Not-Enough-Free-Space screen
## Is your feature request related to a problem? Please describe.
https://www.figma.com/file/vYQD6uZtkPgtmgJ3zfauq0/Zashi-UI?type=design&node-id=3296-20644&mode=design&t=WJtQ9m60zYFindSA-0

### LINKED ISSUE #1410: Add `default: ` before the default server hostname in UI

### PR DESCRIPTION
**Pull request by HonzaR**
- These changes bring the dark theme into the entire app UI
- All screens and UI widgets are now properly colored for both Light and Dark theme
- The PR also brings a few other mostly minor UI changes to simplify the dark mode adoption
- Closes #1011
- Closes https://github.com/Electric-Coin-Company/zashi/issues/32
- Closes #1410
- Closes #1337

> **Note**
>This code review checklist is intended to serve as a starting point for the author and reviewer, although it may not be appropriate for all types of changes (e.g. fixing a spelling typo in documentation).  For more in-depth discussion of how we think about code review, please see [Code Review Guidelines](../blob/main/docs/CODE_REVIEW_GUIDELINES.md).

# Author

- [x] **Self-review** your own code in GitHub's web interface[^1]
- [ ] Add **automated tests** as appropriate
- [ ] Update the [**manual tests**](../blob/main/docs/testing/manual_testing)[^2] as appropriate
- [ ] Check the **code coverage**[^3] report for the automated tests
- [x] Update **documentation** as appropriate (e.g [**README.md**](../blob/main/README.md), [**Architecture.md**](../blob/main/docs/Architecture.md), [**CHANGELOG.md**](../blob/main/CHANGELOG.md), etc.)
- [x] **Run the app** and try the changes
- [x] Pull in the latest changes from the **main** branch and **squash** your commits before assigning a reviewer[^4]

> **Note**
> It is good practice to provide before and after UI  **screenshots** in the description of this PR. This is only applicable for changes that modify the UI.

# Reviewer

- [x] Check the code with the [Code Review Guidelines](../blob/main/docs/CODE_REVIEW_GUIDELINES.md) **checklist**
- [ ] Perform an **ad hoc review**[^5]
- [ ] Review the **automated tests**
- [ ] Review the **manual tests**
- [ ] Review the **documentation**, [**README.md**](../blob/main/README.md), [**Architecture.md**](../blob/main/docs/Architecture.md), etc. as appropriate
- [ ] **Run the app** and try the changes[^6]

[^1]: _Code often looks different when reviewing the diff in a browser, making it easier to spot potential bugs._
[^2]: _While we aim for automated testing of the application, some aspects require manual testing. If you had to manually test something during development of this pull request, write those steps down._
[^3]: _While we are not looking for perfect coverage, the tool can point out potential cases that have been missed. Code coverage can be generated with: `./gradlew check` for Kotlin modules and `./gradlew connectedCheck -PIS_ANDROID_INSTRUMENTATION_TEST_COVERAGE_ENABLED=true` for Android modules._
[^4]: _Having your code up to date and squashed will make it easier for others to review. Use best judgement when squashing commits, as some changes (such as refactoring) might be easier to review as a separate commit._
[^5]: _In addition to a first pass using the code review guidelines, do a second pass using your best judgement and experience which may identify additional questions or comments. Research shows that code review is most effective when done in multiple passes, where reviewers look for different things through each pass._
[^6]: _While the CI server runs the app to look for build failures or crashes, humans running the app are more likely to notice unexpected log messages, UI inconsistencies, or bad output data. Perform this step last, after verifying the code changes are safe to run locally._

### CODE REVIEW

**Review by LukasKorba — APPROVED** [review]
utACK

**Review by Milan-Cerovsky — CHANGES_REQUESTED** [review]
Good PR 👍  Got a few minor comments. Since it's a huge one I'll do a second review too.

File: ui-design-lib/src/main/java/co/electriccoin/zcash/ui/design/component/Checkbox.kt [hunk 1]
```diff
@@ -1,76 +1,122 @@
...
+            }
+        }
+    }
+}
+
+@Preview
+@Composable
+private fun LabeledCheckboxDarkPreview() {
+    ZcashTheme(forceDarkMode = true) {
+        BlankSurface {
```
[the remark this answers is not in the document]
Comment (HonzaR) [reply ui-design-lib/src/main/java/co/electriccoin/zcash/ui/design/component/Checkbox.kt:47]: That's a good point. We could leverage the Preview `showBackground` and `backgroundColor` for the UI components from the design-lib module. I'm in favor of keeping our Previews from the ui-lib module as they are, as we have blank or grid variants there to align with the final screen look as closely as possible. Filed a follow-up on this change #1477

File: ui-design-lib/src/main/java/co/electriccoin/zcash/ui/design/component/Chip.kt [hunk 2]
```diff
@@ -80,32 +104,35 @@ fun ChipIndexed(
...
+                        ),
+                    shape = RoundedCornerShape(size = ZcashTheme.dimens.regularRippleEffectCorner),
+                )
+                .clickable { onClick() },
+        color = ZcashTheme.colors.primaryColor,
         shadowElevation = ZcashTheme.dimens.chipShadowElevation,
     ) {
         Text(
             text = text,
             style = MaterialTheme.typography.bodyLarge,
```
[the remark this answers is not in the document]
Comment (HonzaR) [reply ui-design-lib/src/main/java/co/electriccoin/zcash/ui/design/component/Chip.kt:130]: We're not using `MaterialTheme.typography` in the client ui-lib module. We're using it only here in the design-lib module, which seems fine to me. What do you think?

**Review by Milan-Cerovsky — APPROVED** [review]

File: ui-design-lib/src/main/java/co/electriccoin/zcash/ui/design/component/Dialog.kt [hunk 3]
```diff
@@ -117,3 +140,54 @@ fun AppAlertDialog(
...
+    outerPaddingValues: PaddingValues =
+        PaddingValues(
+            horizontal = ZcashTheme.dimens.spacingNone,
+            vertical = ZcashTheme.dimens.spacingSmall
+        ),
+) {
+    Button(
+        onClick = onClick,
+        modifier =
+            modifier.then(
```
[the remark this answers is not in the document]
Comment (HonzaR) [reply ui-design-lib/src/main/java/co/electriccoin/zcash/ui/design/component/Dialog.kt:180]: Good catch.

File: ui-lib/src/main/java/co/electriccoin/zcash/ui/screen/warning/AndroidNotEnoughSpace.kt [hunk 4]
```diff
@@ -2,27 +2,85 @@
...
+        walletState = walletState,
+    )
 }
 
 @Composable
-private fun WrapNotEnoughSpace(activity: ComponentActivity) {
-    val storageCheckViewModel by activity.viewModels<StorageCheckViewModel>()
-    val spaceRequiredToContinue by storageCheckViewModel.spaceRequiredToContinueMegabytes.collectAsStateWithLifecycle()
+private fun WrapNotEnoughFreeSpace(
+    context: Context,
```
[the remark this answers is not in the document]
Comment (HonzaR) [reply ui-lib/src/main/java/co/electriccoin/zcash/ui/screen/warning/AndroidNotEnoughSpace.kt:56]: Correct, fixed.

File: ui-lib/src/main/java/co/electriccoin/zcash/ui/screen/warning/view/NotEnoughSpaceView.kt [hunk 5]
```diff
@@ -1,77 +1,193 @@
...
+        )
 
-        Small(
-            text = stringResource(id = R.string.space_required_to_continue, spaceRequiredToContinueMegabytes),
-            textAlign = TextAlign.Center,
-            modifier = Modifier.fillMaxWidth()
+        Spacer(modifier = Modifier.height(ZcashTheme.dimens.spacingDefault))
+
+        PrimaryButton(
+            onClick = onSystemStettings,
```
[the remark this answers is not in the document]
Comment (HonzaR) [reply ui-lib/src/main/java/co/electriccoin/zcash/ui/screen/warning/view/NotEnoughSpaceView.kt:187]: Good catch. Thank you.

File: ui-lib/src/main/java/co/electriccoin/zcash/ui/Navigation.kt [hunk 6]
```diff
@@ -291,11 +300,15 @@ private fun MainActivity.NavigationHome(
             },
     )
 
+    val isEnoughSpace by storageCheckViewModel.isEnoughSpace.collectAsStateWithLifecycle()
+
     val sdkStatus = walletViewModel.walletSnapshot.collectAsStateWithLifecycle().value?.status
 
-    if (Synchronizer.Status.DISCONNECTED == sdkStatus) {
+    if (isEnoughSpace == false) {
+        Twig.info { "Not enough free space" }
```
[the remark this answers is not in the document]
Comment (HonzaR) [reply ui-lib/src/main/java/co/electriccoin/zcash/ui/Navigation.kt:308]: That's a good idea. I just filed an issue for this improvement #1478.

File: ui-lib/src/main/java/co/electriccoin/zcash/ui/Navigation.kt [hunk 7]
```diff
@@ -305,6 +318,7 @@ private fun MainActivity.NavigationHome(
             }
         )
     } else if (ConfigurationEntries.IS_APP_UPDATE_CHECK_ENABLED.getValue(RemoteConfig.current)) {
+        Twig.info { "App update available" }
```
[the remark this answers is not in the document]
Comment (HonzaR) [reply ui-lib/src/main/java/co/electriccoin/zcash/ui/Navigation.kt:321]: As above #1478.

File: ui-lib/src/main/java/co/electriccoin/zcash/ui/screen/warning/AndroidNotEnoughSpace.kt [hunk 8]
```diff
@@ -2,27 +2,85 @@
...
+    goSettings: () -> Unit
+) {
+    val walletViewModel by viewModels<WalletViewModel>()
+
+    val storageCheckViewModel by viewModels<StorageCheckViewModel>()
+
+    val walletState = walletViewModel.walletStateInformation.collectAsStateWithLifecycle().value
+
+    val isEnoughFreeSpace = storageCheckViewModel.isEnoughSpace.collectAsStateWithLifecycle().value
+    if (isEnoughFreeSpace == true) {
```
[the remark this answers is not in the document]
Comment (HonzaR) [reply ui-lib/src/main/java/co/electriccoin/zcash/ui/screen/warning/AndroidNotEnoughSpace.kt:34]: As above #1478.

**Pull request merged**